include/envoy/buffer/buffer.h

```diff
@@ -143,7 +143,8 @@ class Instance {
    * Read from a file descriptor directly into the buffer.
    * @param fd supplies the descriptor to read from.
    * @param max_length supplies the maximum length to read.
-   * @return a tuple with the number of bytes read or -1 if there was an error, and the errno.
+   * @return a tuple with the number of bytes read and the errno. If an error occurs, the number
+   *   of bytes read would indicate -1 and the errno would be non-zero.
```

Comment:
How about
"a tuple with the number of bytes read and the errno. If an error occurred, the number of bytes read would indicate -1 and the errno would be non-zero.  Otherwise if bytes were read, errno should not be used."
?

Reply:
Addressed in e5c8ed1.